### DISCUSSION TITLE
Error from pyart.map.grid_rhi_sweeps

### DISCUSSION THREAD

**Discussion by salbenson** [ARM-DOE/pyart#1646] [Q&A] [answered]
Hi, 
I am reading in an RHI cfradial file and I want to grid it.  These are the lines I am using:

radar=pyart.io.read_cfradial(filename)
grid=pyart.map.grid_rhi_sweeps([radar],
    target_sweeps='None',
    grid_size='auto',
    grid_limits='auto',max_z=12000.0,az_rounding_frac=0.25)

Here is my error
Traceback (most recent call last):
  File "/Users/u0079358/Documents/python/convert_cfradial_to_cartesian_rhi.py", line 49, in <module>
    grid=pyart.map.grid_rhi_sweeps([radar],
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/map/grid_mapper.py", line 1055, in grid_rhi_sweeps
    raise TypeError("'grid_shape' must be of type int or tuple")
TypeError: 'grid_shape' must be of type int or tuple

Can you please tell me what I am doin wrong?
Thanks!
Sally

**Comment by zssherman** [comment]
Hi @salbenson, are you plugging these parameters in a custom function? I see grid_size etc. Whichever parameter is feeding into your grid_shape, I would wrap in parenthesis to make it into a tuple.

**Comment by salbenson** [comment]
Hi @zssherman, thanks for your help.  I am using the function here on this page.  I don't see grid_shape in the list of arguments. Do you mean I should be using ('auto') instead of just 'auto'?

https://arm-doe.github.io/pyart/API/generated/pyart.map.grid_rhi_sweeps.html#pyart.map.grid_rhi_sweeps

**Comment by salbenson** [comment]
I started looking at the code for pyart.map.grid_rhi_sweeps and I now think auto isn't an option for 'grid_size'
I think there is a typo in this line of the documentation where grid_size should be grid_limits.  
max_z (float) – maximum height in grid (only used if grid_size is ‘auto’).

So I used 801 for grid_size:
grid=pyart.map.grid_rhi_sweeps([radar],target_sweeps='None',grid_size=801,
    grid_limits='auto',max_z=12000.0,az_rounding_frac=0.25)

This is the error I am now getting:
Traceback (most recent call last):
  File "/Users/u0079358/Documents/python/convert_cfradial_to_cartesian_rhi.py", line 49, in <module>
    grid=pyart.map.grid_rhi_sweeps([radar],
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/map/grid_mapper.py", line 1061, in grid_rhi_sweeps
    [np.max(radar.get_gate_x_y_z(sweep=sw)[0]) for sw in target_sweeps]
            ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'get_gate_x_y_z'

Any ideas about what I am doing wrong here?
Thanks!

**Comment by zssherman** [comment]
@salbenson Mistake on my part, I was looking at the wrong function grid_from_radars. For the grid_rhi_sweeps, I think there is an issue in the documentation, as grid_size does not take 'auto' in the code but I do see it in the docs for it (ill put in a fix for the docs, good catch!). grid_limits is fine as auto. But for grid_size you would have to provide a shape of how many points within your grid_limits for x, y, and z. A tuple of those three numbers. As a note: this code was originally tested on ARM San Diego experiment, I'll need to double check if it was tested on datasets outside of that.

**Reply by zssherman** [reply]
Saw your new post, I would leave radar as not a list, no brackets when plugging it in to the function as it expects a singular radar object, not within a list.

**Reply by salbenson** [reply]
Ok thanks, I took the brackets off of radar and the program is going further.  Now I am getting another error.  I am thinking that it doesn't like target_sweeps='None' either?

Traceback (most recent call last):
  File "/Users/u0079358/Documents/python/convert_cfradial_to_cartesian_rhi.py", line 49, in <module>
    grid=pyart.map.grid_rhi_sweeps(radar,
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/map/grid_mapper.py", line 1061, in grid_rhi_sweeps
    [np.max(radar.get_gate_x_y_z(sweep=sw)[0]) for sw in target_sweeps]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 498, in get_gate_x_y_z
    azimuths = self.get_azimuth(sweep)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 430, in get_azimuth
    s = self.get_slice(sweep)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 375, in get_slice
    start, end = self.get_start_end(sweep)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 371, in get_start_end
    return self.get_start(sweep), self.get_end(sweep)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 361, in get_start
    self._check_sweep_in_range(sweep)
  File "/Users/u0079358/miniconda3/lib/python3.12/site-packages/pyart/core/radar.py", line 305, in _check_sweep_in_range
    if sweep < 0 or sweep >= self.nsweeps:
       ^^^^^^^^^
TypeError: '<' not supported between instances of 'str' and 'int'

**Comment by zssherman** [comment, marked as answer]
@salbenson Ah yep good catch! Another documentation fail. The code needs a check for None, even though the docs state it does... I'll add that into a fix as well. I would provide a list or the specific sweep for now.

**Reply by zssherman** [reply]
@salbenson I created a PR that should fix the sweep issue and changes the docs as they should be. Let me know if by providing the sweeps you need, fixes your error above.

**Reply by salbenson** [reply]
@zssherman yes, when providing the sweep number I was able to read in the data with no further problem.  Thanks for looking into this.  I appreciate it.

**Reply by zssherman** [reply]
@salbenson Awesome glad that worked!
